pkg/kubecost/summaryallocation.go

```diff
@@ -40,6 +40,10 @@ type SummaryAllocation struct {
 	ExternalCost           float64               `json:"externalCost"`
 	Share                  bool                  `json:"-"`
 	UnmountedPVCost        float64               `json:"-"`
+	// RawAllocationOnly added back in to the summary to support row modals.
+	// RawAllocationOnly is a pointer so if it is not present it will be
+	// marshalled as null rather than as an object with Go default values.
+	RawAllocationOnly *RawAllocationOnlyData `json:"rawAllocationOnly"`
```

Comment:
`RawAllocationOnly` does not belong in `SummaryAllocation`, in my opinion.

RAO is only well-defined on unaccumulated, unaggregated Allocation data (let me know if an explanation would help) while `SummaryAllocation`'s main purpose is to serve accumulated and/or aggregated Allocation data with a small data footprint.

Why do you need this data here? If I'm remembering the feature request correctly, you should be able to get a per-request sizing-row max by calling the request sizing API with target utilization = 100%.

If you need per-container-instance (i.e. per pod/container tuple) max, please use the `/allocation` API with filters as it should already be able to serve that data and is designed for maximum-resolution information.

Reply:
I got that request from James--it sounds like the row modal is doing a summary allocation request, and the missing items from the API are the RawAllocationOnly and Properties fields, for him to populate the row.

Reply:
Given what you're saying about better API calls then, @jjarrett21 would it be better to swap to a different call for populating those modals, rather than adding the requisite information into the SummaryAllocations?

Reply:
From conversation with Austin, it's sounding like there's an alternate path that wouldn't necessitate these changes to SummaryAllocation; if that's the case, then we can probably just close this out. I'm holding on confirmation